src/main/java/org/broadinstitute/hellbender/utils/codecs/GENCODE/GencodeGtfFeature.java

```diff
@@ -0,0 +1,1320 @@
+        antisense,
+
+        known_ncrna,
+
+        // Have homology to proteins but generally suffer from a disrupted coding sequence and an active homologous gene can be found at another locus. Sometimes these entries have an intact coding sequence or an open but truncated ORF, in which case there is other evidence used (for example genomic polyA stretches at the 3' end) to classify them as a pseudogene. Can be further classified as one of the following.
+        pseudogene,
+
+        // Pseudogene that lack introns and is thought to arise from reverse transcription of mRNA followed by reinsertion of DNA into the genome.
+        processed_pseudogene,
+
+        // Pseudogene owing to a SNP/DIP but in other individuals/haplotypes/strains the gene is translated.
+        polymorphic_pseudogene,
+
+        // Pseudogene owing to a reverse transcribed and re-inserted sequence.
+        retrotransposed,
+
+        // Pseudogene where protein homology or genomic structure indicates a pseudogene, but the presence of locus-specific transcripts indicates expression.
+        transcribed_processed_pseudogene,
+        transcribed_unprocessed_pseudogene,
+        transcribed_unitary_pseudogene,
+
+        // Pseudogene that has mass spec data suggesting that it is also translated.
+        translated_processed_pseudogene,
+        translated_unprocessed_pseudogene,
+
+        // A species specific unprocessed pseudogene without a parent gene, as it has an active orthologue in another species.
+        unitary_pseudogene,
+
+        // Pseudogene that can contain introns since produced by gene duplication.
+        unprocessed_pseudogene,
+
+        // Used to tag mistakes in the public databases (Ensembl/SwissProt/Trembl)
+        artifact,
+
+        // Long, intervening noncoding (linc) RNA that can be found in evolutionarily conserved, intergenic regions.
+        lincRNA,
+
+        // Unspliced lncRNA that is several kb in size.
+        macro_lncRNA,
+
+        // Transcript where ditag and/or published experimental data strongly supports the existence of short non-coding transcripts transcribed from the 3'UTR.
+        three_prime_overlapping_ncRNA,
+
+        // Otherwise viable coding region omitted from this alternatively spliced transcript because the splice variation affects a region coding for a protein domain.
+        disrupted_domain,
+
+        // Short non coding RNA gene that forms part of the vault ribonucleoprotein complex.
+        vaultRNA,
+
+        // A non-coding locus that originates from within the promoter region of a protein-coding gene, with transcription proceeding in the opposite direction on the other strand.
+        bidirectional_promoter_lncRNA;
+
+        public static GeneTranscriptType getEnum(String s) {
+
+            if (s.startsWith("3")) {
+                s = "three_" + s.substring(1);
```

Comment:
Yeah - the enums were kind of annoying.  I've refactored them to be more uniform and slightly less bad.